### PR TITLE
Use all requested setting backends again [cont]

### PR DESCRIPTION
**Pull request by swick**
Based on https://github.com/flatpak/xdg-desktop-portal/pull/1536

### CODE REVIEW

File: src/xdp-portal-impl.c [hunk 1]
```diff
@@ -622,6 +629,51 @@ find_portal_implementation_iface (const PortalInterface *iface)
...
+
+  for (size_t i = 0; iface->portals && iface->portals[i]; i++)
+    {
+      const char *portal = iface->portals[i];
+
+      for (const GList *l = implementations; l != NULL; l = l->next)
+        {
+          XdpPortalImplementation *impl = l->data;
+
+          if (!g_str_equal (impl->source, portal) && !g_str_equal (portal, "*"))
```
Comment (Sodivad) [inline src/xdp-portal-impl.c:650]: If someone has 

`org.freedesktop.impl.portal.Settings=foo;bar;*`

then  what we end up with adding `foo` and `bar` twice

Comment (Sodivad) [reply src/xdp-portal-impl.c:650]: > True, I should add tests for * as well. The reason why I changed the loops is that iterating over the impls in the outer loop destroys the config order

Makes sense.

Comment (swick) [reply src/xdp-portal-impl.c:650]: Took the easy route and just check if the impl is already in the impls list. This is now loops in a loops in a loop but should be fine as the number of items is low.

Also added tests which use `*` and actually test that the backend is only called once with a config of `test1;*`.

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request merged**